### PR TITLE
Store Sierra contracts in starknet

### LINKED ISSUE #71: Save Sierra contract class in Starknet
In [starknet_getClassAt](https://github.com/starkware-libs/starknet-specs/blob/798c29469aba76b204bf6940a8caf0d114bcc64b/api/starknet_api_openrpc.json#L394) we have to return CONTRACT_CLASS which is Sierra contract. Right now we can't fetch Sierra contract since `InMemoryStateReader` keeps only casm classes. `Starknet` struct will need to keep Sierra contracts

### PR DESCRIPTION
**Pull request by taco-paco**
## Development related changes
Now `get_class`, `get_class_at` functionality can be fully added.
Merged impls of `Starknet` into 1.

- Closes #71
- Related (addresses partially) #51

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet/src/starknet/mod.rs [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+            timeout: u16::default(),
+            gas_price: u64::default(),
+            chain_id: StarknetChainId::TestNet,
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct Starknet {
+    pub(in crate::starknet) state: StarknetState,
```
Comment (marioiordanov) [inline crates/starknet/src/starknet/mod.rs:68]: there is no need to be defined as pub(in crate::starknet) it could be pub(crate) its the same

**Review by marioiordanov — CHANGES_REQUESTED** [review]
IMO access modifiers should be changed.

File: crates/starknet/src/starknet/mod.rs (hunk 1 above)
Comment (taco-paco) [reply crates/starknet/src/starknet/mod.rs:68]: Those fields initially were declared private by you. This is not needed anymore? I decided to keep that logic restricting access only to friend functions in `starknet` module

Comment (marioiordanov) [reply crates/starknet/src/starknet/mod.rs:68]: Where are the friend functions

Comment (taco-paco) [reply crates/starknet/src/starknet/mod.rs:68]: The fields with pub(in crate::starknet) are accessible only in `starknet` module not crate so those fields aren't accessible in `crate::account` module or any other module

Comment (taco-paco) [reply crates/starknet/src/starknet/mod.rs:68]: `friend` is more like c++ notation that is mimicked with `pub(in some::module)`

Comment (marioiordanov) [reply crates/starknet/src/starknet/mod.rs:68]: okay, that was misunderstanding about the pub(in crate::starknet).

**Review by marioiordanov — APPROVED** [review]

File: crates/starknet/src/error.rs [hunk 2]
```diff
@@ -19,6 +19,8 @@ pub enum Error {
     IoError(#[from] std::io::Error),
     #[error("Error when reading file {path}")]
     ReadFileError { source: std::io::Error, path: String },
+    #[error("Format error")]
+    FormatError,
```
Comment (FabijanC) [inline crates/starknet/src/error.rs:23]: Where is this used? If not used, should it be deleted?

**Review by FabijanC — COMMENTED** [review]
Left one comment and triggered the PR (by slightly modfying README.md)

File: crates/starknet/src/error.rs (hunk 2 above)
Comment (taco-paco) [reply crates/starknet/src/error.rs]: Removed

**Review by FabijanC — APPROVED** [review]

**Pull request merged**